client/src/pages/PlantDetails.jsx

```diff
@@ -0,0 +1,143 @@
+import { useLazyQuery, useMutation } from "@apollo/client";
+import { Delete } from "@mui/icons-material";
+import { Box, Button, CircularProgress, FormControl, IconButton, Paper, Stack, TextareaAutosize, Typography } from "@mui/material";
+import React, { useCallback, useEffect, useState } from "react";
+import { useParams } from "react-router-dom";
+import { ResponsiveImageContainer } from "../components/ResponsiveImageContainer";
+import { QUERY_PLANT } from "../utils/queries";
+import CenteredCircularProgress from "../components/CenteredCircularProgress";
+import { ADD_PLANT_NOTE, DELETE_PLANT_NOTE } from "../utils/mutations";
+
+/**
+ *
+ * @param {*} props
+ * @returns
+ */
+function PlantDetails() {
+  const params = useParams();
+  const [query, { error, loading }] = useLazyQuery(QUERY_PLANT);
+  const [addNoteMutation] = useMutation(ADD_PLANT_NOTE, {
+    onError: (e) => {
+      /* TO DO: add error handling */
+      console.error(e.message);
+      alert("Error adding note: " + e.message);
+    },
+  });
+
+  const [deletePlantNoteMutation] = useMutation(DELETE_PLANT_NOTE, {
+    onError: (e) => {
+      /* TO DO: add error handling */
+      console.error(e.message);
+      alert("Error deleting note: " + e.message);
+    },
+  });
+
+  const [plant, setPlant] = useState(undefined);
+
+  const [notes, setNotes] = useState(undefined);
+
+  const handleAddNote = (e) => {
+    e.preventDefault();
+    setNotes([...(notes ?? []), e.target.elements.note.value]);
+    addNoteMutation({
+      variables: { id: params.id, note: e.target.elements.note.value },
+      refetchQueries: [{ query: QUERY_PLANT, variables: { id: params.id } }],
+    })
+      .then(() => loadPlants())
+      .finally(() => e.target.reset());
+  };
+
+  const loadPlants = useCallback(async () => {
+    query({ variables: { id: params.id } }).then((res) => {
+      console.log("plant updated:", res.data.plant);
+      setPlant(res.data.plant);
+    });
+  }, [query, params]);
+
+  const deletePlantNote = (id, noteId) => {
+    console.log("delete plant note", noteId);
+    deletePlantNoteMutation({ variables: { id, noteId }, refetchQueries: [{ query: QUERY_PLANT, variables: { id: params.id } }] }).then(
+      () => loadPlants()
+    );
+  };
+
+  useEffect(() => {
```

Comment:
useEffect missing dependencies, should not return jsx, and conditionals shouldn't be tied to hooks.